extensions/amp-pinterest/0.1/follow-button.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Util} from './util';
+
+// Popup options
+const POP_FOLLOW = `status=no,resizable=yes,scrollbars=yes,
+  personalbar=no,directories=no,location=no,toolbar=no,
+  menubar=no,width=1040,height=640,left=0,top=0`;
+
+/*
+ * Pinterest Follow Button
+ * @attr data-href:  the url of the user's profile to follow
+ * @attr data-label: the text to display (user's full name)
+ **/
+export class FollowButton {
+
+  constructor(rootElement) {
+    AMP.assert(rootElement.getAttribute('data-href'),
+      'The data-href attribute is required for follow buttons');
+    AMP.assert(rootElement.getAttribute('data-label'),
+      'The data-label attribute is required for follow buttons');
+    this.element = rootElement;
+  }
+
+  /*
+   * Override the default href click handling to log and open popup
+   * @param {Event} evt: the HTML event object
+   **/
+  handleClick(evt) {
+    evt.preventDefault();
+    window.open(this.href, 'pin' + new Date().getTime(), POP_FOLLOW);
+    Util.log(`&type=button_follow&href=${this.href}`);
+  }
+
+  /*
+   * Convert a string into a DOM node
+   * @param {String} str: the string to convert
+   * @returns {Element}
+   **/
+  stringToDOM(str) {
+    const temp = document.createElement('span');
+    temp/*REVIEW*/.innerHTML = str;
```

Comment:
There really is no chance of XSS since the only string values are developer input, and Util.make doesn't quite do the same thing. I also lean towards readability, which this format definitely provides over Util.make. I'd prefer to move the Pin widget, which is a much larger template, to use the string template format for readability as well, but if you're absolutely against it I'm happy to comply and remove these.